homeassistant/components/neato/__init__.py

```diff
@@ -103,10 +103,12 @@ async def async_setup_entry(hass, entry):
         await hass.async_add_executor_job(hub.update_robots)
     except NeatoRobotException:
         _LOGGER.debug("Failed to connect to Neato API")
-        return False
+        raise ConfigEntryNotReady
+
+    hass.data[NEATO_LOGIN] = hub
 
     for component in ("camera", "vacuum", "switch", "sensor"):
-        hass.async_create_task(
+        hass.async_add_job(
```

Comment:
`hass.async_add_job` is legacy for general use. We should keep `hass.async_create_task`.

Reply:
@MartinHjelmare ok will submit a PR.  I changed it because when I check the docs `hass.async_add_job` is listed and I thought it was incorrect.

https://developers.home-assistant.io/docs/config_entries_index/#for-platforms

Do we need this example updated too? Or did I not understand it fully? :)

Reply:
The example is stale and should be updated. :+1:

Reply:
Done https://github.com/home-assistant/core/pull/37913

and done :) https://github.com/home-assistant/developers.home-assistant/pull/604